packages/hoppscotch-sh-admin/src/composables/useConfigHandler.ts

```diff
@@ -212,8 +149,57 @@ export function useConfigHandler(updatedConfigs?: ServerConfigs) {
     );
   };
 
+  /**
+   * The updated configs are transformed into a format that can be used by the mutations
+   */
+
+  const workingUpdatedConfigs: ConfigTransform[] = [
+    {
+      config: GOOGLE_CONFIGS,
+      enabled: updatedConfigs?.providers.google.enabled,
+      field: updatedConfigs?.providers.google.fields,
+    },
+    {
+      config: GITHUB_CONFIGS,
+      enabled: updatedConfigs?.providers.github.enabled,
+      field: updatedConfigs?.providers.github.fields,
+    },
+    {
+      config: MICROSOFT_CONFIGS,
+      enabled: updatedConfigs?.providers.microsoft.enabled,
+      field: updatedConfigs?.providers.microsoft.fields,
+    },
+    {
+      config: MAIL_CONFIGS,
+      enabled: updatedConfigs?.mailConfigs.enabled,
+      field: updatedConfigs?.mailConfigs.fields,
+    },
+  ];
+
+  // Push or filter the configs based on the enabled condition
+  const transformInfraConfigs = (workingConfigs: ConfigTransform[]) => {
+    let newConfigs: UpdatedConfigs[] = [];
+
+    workingConfigs.forEach(({ config, enabled, field }) => {
+      config.forEach(({ name, key }) => {
+        if (enabled && field) {
+          const value = typeof field === 'string' ? field : String(field[key]);
+          newConfigs.push({ name, value });
+        } else {
+          newConfigs = newConfigs.filter((item) => item.name !== name);
+        }
```

Comment:
Could you elaborate on the need for this filtering logic (`else` block)? Since this comes into play only while the user opts to save the modified configuration and `newConfigs` is maintained locally, conditionally adding to the list (`if` block alone) would do, right?